src/pdf/ShoppingList/shoppingListPdfDataProps.ts

```diff
@@ -89,10 +90,11 @@ export const prepareItemsListForHousehold = async (
         if (listItemError) {
             return { data: null, error: listItemError };
         }
-        itemsList.push({
-            description: row.item_name,
-            ...listItemData,
-        });
+        row.list_type === listType &&
```

Comment:
Can we move this before all the processing, so the check is the first thing in the for loop and we don't do any processing if we don't need to? We can use the "Early return pattern" (ask Joel and Leah) and a `continue` statement, so it'll look like this:
```
for (const row of listData) {
        if (row.list_type !== listType) {
            continue
        }
        const { data: listItemData, error: listItemError } = await getQuantityAndNotes(
            row,
            householdSize
        );
        if (listItemError) {
            return { data: null, error: listItemError };
        }
        itemsList.push({
            description: row.item_name,
            ...listItemData,
        });
    }
```